artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -174,6 +184,8 @@ public class ServerSessionPacketHandler implements ChannelHandler {
 
    private final Pool<NullResponseMessage_V2> poolNullResponseV2;
 
+   private final Map<Integer, String> producers = new IntObjectHashMap<>();
```

Comment:
This should have been declared as:

private final IntObjectHashMap<String> producers = new IntObjectHashMap<>();

if you keep the Map declaration you will be using boxing every time an integer is used.


not a big deal though.. I'm amending myself during merge.. just a FYI.